8.8/apache-buster/Dockerfile

```diff
@@ -57,17 +56,23 @@ RUN { \
 		echo 'opcache.fast_shutdown=1'; \
 	} > /usr/local/etc/php/conf.d/opcache-recommended.ini
 
-WORKDIR /var/www/html
-
 # https://www.drupal.org/node/3060/release
 ENV DRUPAL_VERSION 8.8.8
-ENV DRUPAL_MD5 c8265f0228ca0306de7b2755e4731a3d
 
+# https://github.com/drupal/drupal/blob/9.0.1/composer.lock#L4052-L4053
+COPY --from=composer:1.10 /usr/bin/composer /usr/local/bin/
 RUN set -eux; \
-	curl -fSL "https://ftp.drupal.org/files/projects/drupal-${DRUPAL_VERSION}.tar.gz" -o drupal.tar.gz; \
-	echo "${DRUPAL_MD5} *drupal.tar.gz" | md5sum -c -; \
-	tar -xz --strip-components=1 -f drupal.tar.gz; \
-	rm drupal.tar.gz; \
-	chown -R www-data:www-data sites modules themes
+	export COMPOSER_HOME="$(mktemp -d)"; \
+	cd /var/www/; \
+	# composer won't install to a non-empty dir: 'Project directory "/var/www/" is not empty.'
+	rmdir /var/www/html; \
+	composer create-project --no-install "drupal/recommended-project:$DRUPAL_VERSION" ./; \
+	sed -i 's!web/!html/!g' composer.json; \
+	composer install; \
```

Comment:
Do you know whether there's a flag/variable/etc to tell `composer` to completely fail if it can't satisfy what's in the lock file, or is that the default behavior of `composer install`? (IIRC, `bundle` and `npm` have something like that, but I'm not sure whether `composer` does.)

Reply:
It seems that is the default.

>Process Exit Codes
>- 0: OK
>- 1: Generic/unknown error code
>- 2: Dependency solving error code
>
> \- https://getcomposer.org/doc/03-cli.md#process-exit-codes


 Looking at the `install` help text, I should at least add `--no-interaction` (just in case).

```console
root@5c9880846776:/var/www# composer install --help
Usage:
  install [options] [--] [<packages>]...
  i

Arguments:
  packages                       Should not be provided, use composer require instead to add a given package to composer.json.

Options:
      --prefer-source            Forces installation from package sources when possible, including VCS information.
      --prefer-dist              Forces installation from package dist even for dev versions.
      --dry-run                  Outputs the operations but will not execute anything (implicitly enables --verbose).
      --dev                      Enables installation of require-dev packages (enabled by default, only present for BC).
      --no-dev                   Disables installation of require-dev packages.
      --no-custom-installers     DEPRECATED: Use no-plugins instead.
      --no-autoloader            Skips autoloader generation
      --no-scripts               Skips the execution of all scripts defined in composer.json file.
      --no-progress              Do not output download progress.
      --no-suggest               Do not show package suggestions.
  -o, --optimize-autoloader      Optimize autoloader during autoloader dump
  -a, --classmap-authoritative   Autoload classes from the classmap only. Implicitly enables `--optimize-autoloader`.
      --apcu-autoloader          Use APCu to cache found/not-found classes.
      --ignore-platform-reqs     Ignore platform requirements (php & ext- packages).
  -h, --help                     Display this help message
  -q, --quiet                    Do not output any message
  -V, --version                  Display this application version
      --ansi                     Force ANSI output
      --no-ansi                  Disable ANSI output
  -n, --no-interaction           Do not ask any interactive question
      --profile                  Display timing and memory usage information
      --no-plugins               Whether to disable plugins.
  -d, --working-dir=WORKING-DIR  If specified, use the given directory as working directory.
      --no-cache                 Prevent use of the cache
  -v|vv|vvv, --verbose           Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

Help:
  The install command reads the composer.lock file from
  the current directory, processes it, and downloads and installs all the
  libraries and dependencies outlined in that file. If the file does not
  exist it will look for composer.json and do the same.
  
  php composer.phar install
  
  Read more at https://getcomposer.org/doc/03-cli.md#install-i
```